File: traversal/src/test/scala/overflowdb/traversal/testdomains/gratefuldead/FollowedBy.scala
package overflowdb.traversal.testdomains.gratefuldead

import overflowdb._
import scala.jdk.CollectionConverters._

class FollowedBy(graph: Graph, outVertex: NodeRef[SongDb], inVertex: NodeRef[SongDb])
    extends Edge(graph, FollowedBy.Label, outVertex, inVertex, FollowedBy.PropertyNames.all.asJava) {
  def weight: Option[Integer] =
    Option(property(FollowedBy.Properties.Weight))
}

object FollowedBy {
  val Label = "followedBy"

  object Properties {
    val Weight = new PropertyKey[Int]("weight")
  }

  object PropertyNames {
    val Weight = "weight"
    val all: Set[String] = Set(Weight)
  }

  val layoutInformation = new EdgeLayoutInformation(Label, PropertyNames.all.asJava)
  var factory: EdgeFactory[FollowedBy] = new EdgeFactory[FollowedBy] {
    override def forLabel(): String = FollowedBy.Label

    override def createEdge(graph: Graph, outNode: NodeRef[NodeDb], inNode: NodeRef[NodeDb]): FollowedBy =
      new FollowedBy(graph, outNode.asInstanceOf[NodeRef[SongDb]], inNode.asInstanceOf[NodeRef[SongDb]])
  }
}

File: core/src/test/java/overflowdb/testdomains/gratefuldead/FollowedBy.java
package overflowdb.testdomains.gratefuldead;

import overflowdb.EdgeFactory;
import overflowdb.EdgeLayoutInformation;
import overflowdb.NodeRef;
import overflowdb.Graph;
import overflowdb.Edge;
import overflowdb.PropertyKey;

import java.util.Arrays;
import java.util.HashSet;

public class FollowedBy extends Edge {
  public static final String LABEL = "followedBy";
  public static final String WEIGHT = "weight";
  public static final PropertyKey<Integer> WEIGHT_KEY = new PropertyKey<>(WEIGHT);
  public static final HashSet<String> PROPERTY_KEYS = new HashSet<>(Arrays.asList(WEIGHT));

  public Integer weight() {
    return (Integer) property(WEIGHT);
  }

  public FollowedBy(Graph graph, NodeRef outVertex, NodeRef inVertex) {
    super(graph, LABEL, outVertex, inVertex, PROPERTY_KEYS);
  }

  public static final EdgeLayoutInformation layoutInformation = new EdgeLayoutInformation(LABEL, PROPERTY_KEYS);

  public static EdgeFactory<FollowedBy> factory = new EdgeFactory<FollowedBy>() {
    @Override
    public String forLabel() {
      return FollowedBy.LABEL;
    }

    @Override
    public FollowedBy createEdge(Graph graph, NodeRef outVertex, NodeRef inVertex) {
      return new FollowedBy(graph, outVertex, inVertex);
    }
  };
}
